test/components/submission/list.spec.js

```diff
@@ -489,4 +437,72 @@ describe('SubmissionList', () => {
       });
     });
   });
+
+  describe('$select query parameter', () => {
+    // Create a form with 20 string fields. The first string field is in a
+    // group.
+    beforeEach(() => {
+      const { group, string } = testData.fields;
+      const fields = [group('/g'), string('/g/s1')];
+      for (let i = 2; i <= 20; i += 1) fields.push(string(`/s${i}`));
+      testData.extendedForms.createPast(1, { fields });
+    });
+
+    const $select = (url) => relativeUrl(url).searchParams.get('$select');
+
+    it('does not specify $select for the first chunk of submissions', () =>
+      loadSubmissionList().beforeEachResponse((_, { url }) => {
+        if (url.includes('.svc/Submissions')) should.not.exist($select(url));
+      }));
+
+    it('specifies $select for the second chunk', () => {
+      testData.extendedSubmissions.createPast(2);
+      return loadSubmissionList({
+        props: { top: () => 1 }
+      })
+        .complete()
+        .request(scroll)
+        .beforeEachResponse((_, { url }) => {
+          $select(url).should.equal('__id,__system,g/s1,s2,s3,s4,s5,s6,s7,s8,s9,s10');
+        })
+        .respondWithData(() => testData.submissionOData(1, 1));
+    });
+
+    it('specifies $select if the refresh button is clicked', () =>
+      loadSubmissionList()
+        .complete()
+        .request(component =>
+          component.get('#submission-list-refresh-button').trigger('click'))
+        .beforeEachResponse((_, { url }) => {
+          $select(url).should.equal('__id,__system,g/s1,s2,s3,s4,s5,s6,s7,s8,s9,s10');
+        })
+        .respondWithData(() => testData.submissionOData(1, 0)));
+
+    it('specifies $select if a filter is changed', () =>
+      load('/projects/1/forms/f/submissions', { attachTo: document.body })
```

Comment:
This test uses `load()`, mounting `App` and injecting a real router, instead of using `loadSubmissionList()`. That's because changing the filter will change the route.